src/mlpack/core/hpt/hpt_impl.hpp

```diff
@@ -0,0 +1,234 @@
+/**
+ * @file hpt_impl.hpp
+ * @author Kirill Mishchenko
+ *
+ * Implementation of hyper-parameter tuning.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_HPT_HPT_IMPL_HPP
+#define MLPACK_CORE_HPT_HPT_IMPL_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace hpt {
+
+template<typename MLAlgorithm,
+         typename Metric,
+         template<typename, typename, typename, typename, typename> class CV,
+         typename Optimizer,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+template<typename... CVArgs>
+HyperParameterTuner<MLAlgorithm,
+                    Metric,
+                    CV,
+                    Optimizer,
+                    MatType,
+                    PredictionsType,
+                    WeightsType>::HyperParameterTuner(const CVArgs&... args) :
+    cv(args...), relativeDelta(0.01), minDelta(1e-10) {}
```

Comment:
Should we provide a default constructor and set the default vales there? That way doxygen can pick up the default values and we can also add a note that the values are not necessarily good for the given problem, and suggest to adjust each to the task at hand. Let me know what you think.

Reply:
It is problematic since we need to accept variadic template parameters in the constructor. I thought about mentioning the default values in the documentation for the getters and setters, but I hadn't seen anything like that in the mlpack codebase. Let me know if you think it's ok to do that.

Reply:
Right I missed that, I think you could add a note to the class description itself, but adding it to the getter/setter documentation is also fine; it's up to you.